src/common/utils.cpp

```diff
@@ -52,28 +52,42 @@ std::string getDBusServiceName(sdbusplus::bus::bus& bus,
 
     // In OpenBMC, the object path will be hosted by a single service
     // i.e more than one service cannot host the same object path.
-    // Note that for legacy reasons, phosphor-state-manager registers two
-    // service names, xyz.openbmc_project.State.Host and
-    // xyz.openbmc_project.State.Host0. This was to support multi-host
-    // designs but also support legacy users. This is the one exception
-    // to the "more than one service" rule
-    if ((servicesName.size() > 1) &&
-        (servicesName[0].first.find("xyz.openbmc_project.State.Host") ==
-         std::string::npos))
+    if (servicesName.size() > 1)
     {
-        std::string serviceNameList{""};
-
-        std::for_each(servicesName.begin(), servicesName.end(),
-                      [&serviceNameList](const auto& serviceName) {
-                          serviceNameList.append(serviceName.first + ",");
-                      });
-
-        log<level::ERR>(fmt::format("The given object path hosted by "
-                                    "more than one services [{}]",
-                                    serviceNameList)
-                            .c_str());
-        throw std::runtime_error(
-            "Given object path hosted by more than one service");
+        // Note that for legacy reasons, phosphor-state-manager registers two
+        // service names, (xyz.openbmc_project.State.Host and
+        // xyz.openbmc_project.State.Host0) or
+        // (xyz.openbmc_project.State.Chassis and
+        // xyz.openbmc_project.State.Chassis0). This was to support multi-host
+        // designs but also support legacy users. This is the one exception
+        // to the "more than one service" rule
+        auto isExceptionService = [](const auto& serviceName) {
+            std::set<std::string> exceptionServices(
+                {"xyz.openbmc_project.State.Host",
+                 "xyz.openbmc_project.State.Chassis"});
+
+            return exceptionServices.contains(serviceName);
+        };
+
+        if (!isExceptionService(servicesName[0].first))
```

Comment:
This is not going to work if the 0th index contains `xyz.openbmc_project.State.Host0` or `xyz.openbmc_project.State.Chassis0`

I think we should loop with all returned service names and return that service name if it is found in the exceptional service list.

Reply:
With Ln. number 103, I assumed that Always the service name xyz.openbmc_project.State.Host always comes first and so we dont need a loop. But as you pointed it out and it might be better for future, ran a loop and return that service name if it matched. This solution should work for all cases.